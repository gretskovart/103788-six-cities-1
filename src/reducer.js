import offers from './mocks/offers';

const getPlaces = (selectedCity = `Amsterdam`) => offers.filter((it) => it.city === selectedCity);

const initialState = {
  activeCity: `Amsterdam`,
  offers: getPlaces()
};

const actionCreators = {
  changeCity: (selectedCity) => {
    return ({
      type: `changeCity`,
      payload: selectedCity
    });
  },
  getOffers: (selectedCity) => {
    const places = getPlaces(selectedCity);

    return ({
      type: `getOffers`,
      payload: places
    });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `changeCity`:
      return Object.assign({}, state, {
        activeCity: action.payload,
      });
    case `getOffers`:
      return Object.assign({}, state, ({
        offers: action.payload
      }));
  }

  return state;
};

export {actionCreators, reducer};
