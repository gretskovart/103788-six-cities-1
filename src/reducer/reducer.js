import {actionCreators, reducer, getData} from './data/reducer';

export {actionCreators, reducer, getData};
