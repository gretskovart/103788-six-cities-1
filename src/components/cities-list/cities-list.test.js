import React from 'react';
import CitiesList from './cities-list.jsx';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {reducer} from './../../reducer';

const store = createStore(reducer);

it(`<CitiesList /> renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <CitiesList
            offers={[]}
          />
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
