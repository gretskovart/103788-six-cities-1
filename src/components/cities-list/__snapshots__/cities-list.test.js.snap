// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CitiesList /> renders correctly 1`] = `null`;
